### PR TITLE
Add Lessons for Users

### PR DESCRIPTION
**Pull request by DataDaoDe**
Requirements:
* A user can click through courses, chapters, and lessons to easily navigate around in them.
* When a user clicks on a lesson, he is immediately transported to his current LearningBlock within the lesson. If this is his first start of the lesson then he will be at the first LessonBlock. If the lesson is in progress it will be the next unfinished LearningBlock i.e. the next LearningBlock he has not completed. If he has already finished the lesson, he will see the lesson summary page where he can reset the lesson to do it over again if he so chooses. 
* A user finishes one LessonBlock after another in sequence until the Lesson is completed. He can jump forwards and backwards in the LessonBlocks if he wants but the Lesson will not be completed until he submits an answer for all LessonBlocks.
* Finishing one lesson unlocks the next lesson. If all lessons in a chapter have been finished, then the next chapter is unlocked.

### CODE REVIEW

**Comment by DataDaoDe** [comment]
[QU-0001: Course Lessons Implementation](https://trello.com/c/7FoXVA5l/68-qu-0001-course-lessons-implementation)

File: src/lib/components/Search.svelte [hunk 1]
```diff
@@ -4,20 +4,38 @@
...
 
 	function internalOnChange(e: Event & { currentTarget: HTMLInputElement }) {
 		onChange(e.currentTarget.value);
 	}
 </script>
 
-<form on:submit|preventDefault class="mt-2">
-	<div class="input-group">
+<form on:submit|preventDefault class={`${formClass || ''}`}>
+	<div class={`input-group ${searchIcon ? 'search-icon' : ''}`}>
```
Comment (nathanfaucett) [inline src/lib/components/Search.svelte:17]: we can use class directives here https://svelte.dev/tutorial/classes

File: src/lib/components/Search.svelte [hunk 2]
```diff
@@ -4,20 +4,38 @@
...
+		{#if searchIcon}
+			<i class="bi bi-search" />
+		{/if}
 		<input
 			type="text"
-			class="form-control"
+			class={`form-control ${searchInputClass || ''}`}
```
Comment (nathanfaucett) [inline src/lib/components/Search.svelte:23]: same here

File: src/lib/components/Search.svelte [hunk 3]
```diff
@@ -4,20 +4,38 @@
...
+	export let formClass: string = '';
 
 	function internalOnChange(e: Event & { currentTarget: HTMLInputElement }) {
 		onChange(e.currentTarget.value);
 	}
 </script>
 
-<form on:submit|preventDefault class="mt-2">
-	<div class="input-group">
+<form on:submit|preventDefault class={`${formClass || ''}`}>
```
Comment (nathanfaucett) [inline src/lib/components/Search.svelte:16]: no need for this just class={formClass}

File: src/lib/components/Search.svelte (hunk 2 above)
Comment (DataDaoDe) [reply src/lib/components/Search.svelte:23]: how? `searchInputClass` can be any class given class string.

**Pull request merged**